src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -242,6 +242,7 @@ public override INamedTypeSymbol CreateErrorTypeSymbol(INamespaceOrTypeSymbol co
             bool isSubmission)
         {
             Debug.Assert(options != null);
+            Debug.Assert(!isSubmission || options.ReferencesSupersedeLowerVersions);
```

Comment:
Is the goal to prevent a submission compilation from being created with ```ReferencesSupersedeLowerVersions == false```? Could the caller still use ```CSharpCompilation.CreateScriptCompilation(options).WithOptions(options.WithReferencesSupersedeLowerVersions(true))```? If so, consider throwing in the `CSharpCompilation..ctor` instead.